### DISCUSSION TITLE
artillery, playwright, fargate how to run a single time script in setup

### DISCUSSION THREAD

**Discussion by zoe-jobson-IL** [artilleryio/artillery#2598] [Artillery]
At present we have a manual script to clear out previous run data by hand before running one of our load tests again, it would be nice if we could run it as part of the test itself via an api, the main issue is that when we run with multiple fargate pods, we only want to run it once (NOT once per pod), and for none of the other fargate pods to start running test code until its done. Is there a simple way of doing this.

**Comment by hassy** [comment]
Hi @zoe-jobson-IL 👋 There are two ways right now:

1. Run a clean-up script before running Artillery
2. Run the clean-up code in a [`before` hook](https://www.artillery.io/docs/reference/test-script#before-and-after-sections) and use some external state storage mechanism (e.g. Redis or DynamoDB) to make sure that only one instance of it is running. The code in the `before` hook would either:
   1. Acquire a lock, run clean-up, and then signal that clean-up is done, or
   2. Wait until "done" signal is received

This is a very common use-case, we're working on an end-to-end example using Redis.
